src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int INCREASE_VALUE = 2;
+    private int size;
+    private float loadFactor;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resizeIfNeeded();
+        Node<K, V> node = new Node<>(key, value);
+        int indexBucket = getIndex(key);
+        if (table[indexBucket] == null) {
+            table[indexBucket] = node;
+        } else {
+            Node<K, V> current = table[indexBucket];
+            Node<K, V> prev = null;
+            while (current != null) {
+                if (Objects.equals(current.key, node.key)) {
+                    current.value = node.value;
+                    return;
+                }
+                prev = current;
+                current = current.next;
+            }
+            prev.next = node;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int indexBucket = getIndex(key);
+        if (table[indexBucket] != null) {
+            Node<K, V> current = table[indexBucket];
+
+            while (current != null) {
+                if (Objects.equals(current.key, key)) {
+                    return current.value;
+                }
```

Comment:
![image](https://github.com/mate-academy/jv-hash-map/assets/63095184/a64610f7-99f0-4a7d-accd-6e31f1e20290)
Can you combine this logic and make separate private method